File: Max_Increase_to_Keep_City_Skyline/Max_Increase_to_Keep_City_Skyline.py
grid=[[3,0,8,4],[2,4,5,7],[9,2,6,3],[0,3,1,0]]


def maxIncreaseKeepingSkyline(grid):
    row = [0]*len(grid)
    col = [0]*len(grid)
    for i in range(len(grid)):
        for j in range(len(grid)):
            row[i] = max(row[i],grid[i][j])
            col[j] = max(col[j],grid[i][j])
            
    out=0
    for i in range(len(grid)):
        for j in range(len(grid)):
            out+=min(col[j],row[i])-grid[i][j]
    return out


print(maxIncreaseKeepingSkyline(grid))

File: Max_Increase_to_Keep_City_Skyline/Max_Increase_to_Keep_City_Skyline.rs
use std::cmp;
fn main(){
    let grid:Vec<Vec<i32>> = vec![vec![3,0,8,4],vec![2,4,5,7],vec![9,2,6,3],vec![0,3,1,0]];
    println!("{} ",maxIncreaseKeepingSkyline(grid));
}

fn maxIncreaseKeepingSkyline(grid:Vec<Vec<i32>>) -> i32{
    let mut col:Vec<i32> = vec![0;grid.len()];
    let mut row:Vec<i32> = vec![0;grid.len()];
    
    println!("{:?} ",col);
    for i in 0..grid.len(){
        for j in 0..grid.len(){
            row[i] = cmp::max(row[i],grid[i][j]);
            col[j] = cmp::max(col[j],grid[i][j]);
        }
    }

    let mut out = 0;
    for x in 0..grid.len(){
        for y in 0..grid.len(){
            out += cmp::min(row[x],col[y]) - grid[x][y] as i32;
        }
    }


    return out;
}